src/power_grid_model/validation/errors.py

```diff
@@ -317,19 +320,19 @@ class InvalidIdError(SingleFieldValidationError):
     """
 
     _message = "Field {field} does not contain a valid {ref_components} id for {n} {objects}. {filters}"
-    ref_components: List[str]
+    ref_components: List[ComponentType]
 
     def __init__(
         self,
-        component: str,
+        component: ComponentType,
         field: str,
         ids: List[int],
-        ref_components: Union[str, List[str]],
+        ref_components: Union[ComponentType, List[ComponentType]],
         filters: Optional[Dict[str, Any]] = None,
     ):
         # pylint: disable=too-many-arguments
         super().__init__(component=component, field=field, ids=ids)
-        self.ref_components = [ref_components] if isinstance(ref_components, str) else ref_components
+        self.ref_components = [ref_components] if isinstance(ref_components, str) else ref_components  # type: ignore
```

Comment:
Will this make the warning disappear?

```suggestion
        self.ref_components = [ref_components] if isinstance(ref_components, (str, ComponentTypes)) else ref_components  # type: ignore
```

Reply:
isinstance() doesn't work directly with a datatype such as ComponentTypes, hence why I added it.

Reply:
@nitbharambe Never mind, you were correct. It's been fixed.